### PR TITLE
Captain

### PR DESCRIPTION
**Pull request by zeeev**
1. Adding the main runner program
 @edawson , looks like you have some more to do in classify, I can help with that tomorrow. I think we are to the point where w e can say which SVs are in which sample?

2. Flowing @fritzsedlazeck and @pb-cdunn's spaced kmers 

@pb-cdunn can you check that I've added the logic correctly for when to call it?

### CODE REVIEW

File: src/nibpkg/captain.nim [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import compose
+import refmers
+import svidx
+import strformat
+import mainLookup
+import classify
+
+proc all_aboard*(variants_fn, refSeq_fn, prefix, reads_fn: string, kmerSize: int = 21, spacedSeeds : bool = false, space: int = 50, preIndex : bool = false, flank: int = 100) =
```
Comment (brentp) [inline src/nibpkg/captain.nim]: rename this. :|

File: src/nibpkg/read.nim [hunk 2]
```diff
@@ -8,11 +8,13 @@ type Read* = object
     ## key of svid, count of supporting kmers
     compatible_SVs* : CountTable[uint32]
 
-proc process_read*(s : string, idx : svIdx, k:int=25): Read =
+proc process_read*(s : string, idx : svIdx, k:int=25, spacedSeeds : bool = false, space : int = 50): Read =
     # find SVs with kmers intersecting with those from this read.
     var x : Read
-    var l = Dna(s).dna_to_kmers(k)
-    for kmer in l.seeds:
+    var kmers = Dna(s).dna_to_kmers(k)
```
Comment (edawson) [inline src/nibpkg/read.nim:14]: I really like how y'all handled integrating the spaced kmers - nice work.

File: src/nibpkg/read.nim [hunk 3]
```diff
@@ -8,11 +8,13 @@ type Read* = object
     ## key of svid, count of supporting kmers
     compatible_SVs* : CountTable[uint32]
 
-proc process_read*(s : string, idx : svIdx, k:int=25): Read =
+proc process_read*(s : string, idx : svIdx, k:int=25, spacedSeeds : bool = false, space : int = 50): Read =
```
Comment (edawson) [inline src/nibpkg/read.nim:11]: Note, our default kmer size is 25 here and 21 over in ref_kmers. That's going to cause issues.

Can we set a macro in the package somehow that makes the default size consistent across nib?

File: src/nibpkg/captain.nim (hunk 1 above)
Comment (edawson) [reply src/nibpkg/captain.nim]: Could we call it "main?" I would also suggest "run_nib?"

**Pull request merged**